docs/Lua API.rst

```diff
@@ -4415,12 +4415,29 @@ General script API
   command-line. The ``name`` argument should be the name of the script without
   its extension, as would be used on the command line.
 
-  Note that this function lets Lua errors propagate to the caller.
+  Example::
+     dfhack.run_script("armoks-blessing", "all")
+  Scipts that use a dash(``-``) in their arguments still require one when called
+  from ``dfhack.run_script``.
+  Example::
+     dfhack.run_script("gaydar", "-all")
+
+  Unlike ``dfhack.run_command``, ``dfhack.run_script`` allows a different
+  different script to be executed and its value passed as an argument to the main
+  function called by ``dfhack.run_script``.
+
+  Example::
+      dfhack.run_script("some_script", dfhack.gui.getDwarfmodeViewDims());
```

Comment:
Regarding terminology: this is passing in the value of a "Lua expression", not the result of a script (`getDwarfmodeViewDims` is a function, not a script).

There are two reasons I'm recommending against including this:
* You can do this with any Lua function, not just `run_script`. `run_command(someFunctionThatReturnsAString())` is equally valid, for instance.
* It behaves poorly if the arguments aren't a string, as in this example. *Some* scripts might work, but many scripts will not. For example, any script using `processArgs()` will fail if an argument isn't a string:
```
[lua]# dfhack.run_script('teleport', {})
./hack/lua/utils.lua:596: bad argument #1 to 'sub' (string expected, got table)
stack traceback:
	[C]: in function 'string.sub'
	./hack/lua/utils.lua:596: in function 'utils.processArgs'
	../../code/dfhack/scripts/teleport.lua:66: in local 'script_code'
	./hack/lua/dfhack.lua:680: in function 'dfhack.run_script_with_env'
	(...tail calls...)
```

In short: we shouldn't advertise that functions can be called with arguments of the wrong types, because that isn't supposed to work, and it often doesn't work.